nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/ScanHBase.java

```diff
@@ -0,0 +1,564 @@
+            .description("All failed fetches are routed to this relationship.")
+            .build();
+
+    static final String HBASE_TABLE_ATTR = "hbase.table";
+    static final String HBASE_ROWS_COUNT_ATTR = "hbase.rows.count";
+
+    static final List<PropertyDescriptor> properties;
+    static {
+        List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(HBASE_CLIENT_SERVICE);
+        props.add(TABLE_NAME);
+        props.add(START_ROW);
+        props.add(END_ROW);
+        props.add(TIME_RANGE_MIN);
+        props.add(TIME_RANGE_MAX);
+        props.add(LIMIT_ROWS);
+        props.add(REVERSED_SCAN);
+        props.add(BULK_SIZE);
+        props.add(FILTER_EXPRESSION);
+        props.add(COLUMNS);
+        props.add(JSON_FORMAT);
+        props.add(ENCODE_CHARSET);
+        props.add(DECODE_CHARSET);
+        properties = Collections.unmodifiableList(props);
+    }
+
+    static final Set<Relationship> relationships;
+    static {
+        Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_SUCCESS);
+        rels.add(REL_ORIGINAL);
+        rels.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(rels);
+    }
+
+    private volatile Charset decodeCharset;
+    private volatile Charset encodeCharset;
+    private RowSerializer serializer = null;
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        this.decodeCharset = Charset.forName(context.getProperty(DECODE_CHARSET).getValue());
+        this.encodeCharset = Charset.forName(context.getProperty(ENCODE_CHARSET).getValue());
+
+        final String jsonFormat = context.getProperty(JSON_FORMAT).getValue();
+        if (jsonFormat.equals(JSON_FORMAT_FULL_ROW.getValue())) {
+            this.serializer = new JsonFullRowSerializer(decodeCharset, encodeCharset);
+        } else {
+            this.serializer = new JsonQualifierAndValueRowSerializer(decodeCharset, encodeCharset);
+        }
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+
+        final List<ValidationResult> problems = new ArrayList<>();
+
+        final String columns = validationContext.getProperty(COLUMNS).getValue();
+        final String filter = validationContext.getProperty(FILTER_EXPRESSION).getValue();
+
+        if (!StringUtils.isBlank(columns) && !StringUtils.isBlank(filter)) {
+            problems.add(new ValidationResult.Builder()
+                    .subject(FILTER_EXPRESSION.getDisplayName())
+                    .input(filter).valid(false)
+                    .explanation("A filter expression can not be used in conjunction with the Columns property")
+                    .build());
+        }
+
```

Comment:
Should we validate that the min/max time range are both provided or both blank? Based on their description.

Reply:
Since it supports EL, I validate this condition in onTrigger (lines 348-356). If you think we should add it also to customValidate, I can add it